### PR TITLE
mgr/telemetry: add Rook data

### PR DESCRIPTION
**Pull request by yaarith**
Add the first Rook data collection to telemetry's basic channel.

We choose to nag with this collection since we wish to know the volume
of Rook deployments in the wild.

The next Rook collections should have consecutive numbers (basic_rook_v02,
basic_rook_v03, ...).

See tracker below for more details.

Fixes: https://tracker.ceph.com/issues/55740
Signed-off-by: Yaarit Hatuka <yaarit@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/telemetry/module.py [hunk 1]
```diff
@@ -1184,6 +1223,42 @@ def compile_report(self, channels: Optional[List[str]] = None) -> Dict[str, Any]
...
+        except json.decoder.JSONDecodeError:
+            return
+
+        for elem in ROOK_KEYS_BY_COLLECTION:
+            # elem[0] is the full key path (e.g. "rook/node/count/with-csi-nfs-plugin")
+            # elem[1] is the Collection this key belongs to
+            if self.is_enabled_collection(elem[1]):
+                self.add_kv_to_report(report, elem[0], config_kv_dump.get(elem[0]))
+
+    def add_kv_to_report(self, report: Dict[str, object], key_path: str, value: Any) -> None:
```
Comment (ljflores) [inline src/pybind/mgr/telemetry/module.py:1244]: Clever algorithm!

**Review by ljflores — APPROVED** [review]
LGTM. I went through your `add_kv_to_report` algorithm, and it's very clever! Agreed that the rook team should test this on their side to ensure it's working.

**Comment by ljflores** [comment]
@travisn tagging just FYI

**Comment by yaarith** [comment]
Thanks, @ljflores!
Eventually I tested these new changes along with Rook's new changes, since there were issues with accessing the new image I created. The output is here: https://github.com/rook/rook/pull/10341#issuecomment-1144140445

**Comment by ljflores** [comment]
Rados suite results:

http://pulpito.front.sepia.ceph.com/yuriw-2022-06-03_14:09:08-rados-wip-yuri7-testing-2022-06-02-1633-distro-default-smithi/

Failures, unrelated:
    1. https://tracker.ceph.com/issues/55741
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/55808
    4. https://tracker.ceph.com/issues/55853 --> opened a new Tracker for this; unrelated to the PR tested in this run.
    5. https://tracker.ceph.com/issues/55854 --> opened a new Tracker for this; unrelated to the PR tested in this run.
    6. https://tracker.ceph.com/issues/55856 --> opened a new Tracker for this; unrelated to the PR tested in this run.

Details:
    1. cephadm/test_dashboard_e2e.sh: Unable to find element `cd-modal .custom-control-label` when testing on orchestrator/01-hosts.e2e-spec.ts - Ceph - Mgr - Dashboard
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator - Ceph - Orchestrator
    3. task/test_nfs: KeyError: 'events' - Ceph - CephFS
    4. test_cls_rgw.sh: failures in 'cls_rgw.index_list' and 'cls_rgw.index_list_delimited` - Ceph - RGW
    5. Datetime AssertionError in test_health_history (tasks.mgr.test_insights.TestInsights) - Ceph - Mgr
    6. ObjectStore/StoreTest.CompressionTest/2 fails when a collection expects an object not to exist, but it does - Ceph - BlueStore

**Comment by neha-ojha** [comment]
@yaarith we should mention this in the release notes, can be done in a separate PR

**Pull request merged**